ci/regression_test/stress_tests/test_many_tasks.py

```diff
@@ -110,6 +110,45 @@ def method(self, size, *xs):
 stage_3_time = time.time() - start_time
 logger.info("Finished stage 3 in %s seconds.", stage_3_time)
 
+
+#This tests https://github.com/ray-project/ray/issues/10150.
+# The only way to integration test this is via performance. We launch a
+# cluster of 10 nodes w/ 10 cpu's per node. Then we launch 100 tasks. Since
+# the driver submits all these tasks it should easily be able to schedule
+# each task in O(1) iterative spillback queries. If spillback behavior is
+# incorrect, each task will require O(N) queries. Since we limit the number
+# of inflight requests, we will run into head of line blocking and we should
+# be able to measure this timing.
+num_tasks = int(ray.cluster_resources()["GPU"])
+logger.info(f"Scheduling many tasks for spillback.")
+
+@ray.remote(num_gpus=1)
+def func(t):
+    if t%100 == 0:
+        logger.info(f"[spillback test] {t}/{num_tasks}")
+    start = perf_counter()
+    time.sleep(1)
+    end = perf_counter()
+    return start, end, ray.worker.global_worker.node.unique_id
+
+results = ray.get([func.remote(i) for i in range(num_tasks)])
+
+host_to_start_times = defaultdict(list)
+for start, end, host in results:
+    host_to_start_times[host].append(start)
+
+spreads = []
+for host in host_to_start_times:
+    last = max(host_to_start_times[host])
+    first = min(host_to_start_times[host])
+    spread = last - first
+    spreads.append(spread)
+    logger.info(f"Spread: {last - first}\tLast: {last}\tFirst: {first}")
+
+# avg_spread ~ 115 with Ray 1.0 scheduler. ~695 with (buggy) 0.8.7 scheduler.
+avg_spread = sum(spreads)/len(spreads)
```

Comment:
Maybe just report max spread?

Reply:
let's rerun this once the release tests/docker bug is fixed and see what the numbers look like.

i used average instead of min/max because when i tested this w/ cluster_utils on a single node originally, the noise was pretty large, so average was much more consistent than sampling single data points.

Reply:
@barakmich can we run this test with the other release tests? the logs show the individual spreads, so we can derive any statistic we including max this time, then "report" max to stdout in the future if it works.